tests/checkpoints/test_dumb_checkpoint.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+from firedrake import *
+import numpy as np
+
+
+@pytest.fixture(scope="module",
+                params=[False, True],
+                ids=["simplex", "quad"])
+def mesh(request):
+    return UnitSquareMesh(2, 2, quadrilateral=request.param)
+
+
+@pytest.fixture(params=range(1, 4))
+def degree(request):
+    return request.param
+
+
+@pytest.fixture(params=["CG"])
+def fs(request):
+    return request.param
+
+
+@pytest.fixture
+def dumpfile(tmpdir):
+    return str(tmpdir.join("dump.h5"))
+
+
+@pytest.fixture(scope="module")
+def f():
+    m = UnitSquareMesh(2, 2)
+    V = FunctionSpace(m, 'CG', 1)
+    return Function(V, name="f")
+
+
+def run_store_load(mesh, fs, degree, dumpfile):
+
+    V = FunctionSpace(mesh, fs, degree)
+
+    f = Function(V, name="f")
+
+    expr = Expression("x[0]*x[1]")
+
+    f.interpolate(expr)
+
+    f2 = Function(V, name="f")
+
+    dumpfile = op2.MPI.comm.bcast(dumpfile, root=0)
+    chk = DumbCheckpoint(dumpfile, mode=FILE_CREATE)
```

Comment:
The dumpfile is created in a temporary directory using the pytest tmpdir fixture which is guaranteed to be unique, and is cleaned up in due course by pytest.